.github/workflows/ci.yml

```diff
@@ -24,6 +27,17 @@ jobs:
       with:
         go-version: 1.17
 
+      # Don't run this since PR does not have access to secrets. 
+      # Let the make docker-build generate the temp. key
+    - name: Setup signing key
+      if: ${{ github.event_name != 'pull_request' }}
+      run: |
+        echo "$PRIVATE_KEY_PEM" > enclave-config/privatekey.pem
+        sha256sum enclave-config/privatekey.pem
```

Comment:
How does this work? I mean, who consumes the checksum?

Reply:
Nobody is consuming the checksum. It is a visual verification that the right key is deployed. If one has access to the key checksum (outside GitHub) the checksums can be compared and verified that the right key is deployed. 

This can be removed, which makes the debug and verification bit harder.